ir/src/main/scala/org/scalajs/core/ir/Printers.scala

```diff
@@ -399,40 +506,67 @@ object Printers {
             case This()                   => true
             case _                        => false // in particular, Apply
           }
-          if (containsOnlySelectsFromAtom(ctor))
-            print("new ", ctor)
-          else
-            print("new (", ctor, ")")
+          if (containsOnlySelectsFromAtom(ctor)) {
+            print("new ")
+            print(ctor)
+          } else {
+            print("new (")
+            print(ctor)
+            print(')')
+          }
           printArgs(args)
 
         case JSDotSelect(qualifier, item) =>
-          print(qualifier, ".", item)
+          print(qualifier)
+          print(".")
+          print(item)
 
         case JSBracketSelect(qualifier, item) =>
-          print(qualifier, "[", item, "]")
+          print(qualifier)
+          print('[')
+          print(item)
+          print(']')
 
         case JSFunctionApply(fun, args) =>
           fun match {
-            case _:JSDotSelect | _:JSBracketSelect | _:Select =>
-              print("protect(", fun, ")")
+            case _: JSDotSelect | _: JSBracketSelect | _: Select =>
```

Comment:
Coding style: no space after `:` when it's used in a `|`-separated list of alternatives in a `case`.